uportal-war/src/main/java/org/jasig/portal/url/UrlSyntaxProviderImpl.java

```diff
@@ -1091,7 +1099,11 @@ protected void addPortletUrlData(
             final String portletWindowIdStr = portletWindowId.toString();
             prefixedPortletWindowId = SEPARATOR + portletWindowIdStr;
             suffixedPortletWindowId = portletWindowIdStr + SEPARATOR;
-            url.addParameter(PARAM_ADDITIONAL_PORTLET, portletWindowIdStr);
+
+            // handle URI (i.e. search) that requires skipping additional portlet info
+            if (!request.getRequestURI().contains(this.compactParamsForUriFragment)) {
+                url.addParameter(PARAM_ADDITIONAL_PORTLET, portletWindowIdStr);
+            }
```

Comment:
This approach is specific to search.  Is there not a way to determine if there are actual parameter values that would be sent to the additional portlet and if not then skip adding it?  As I recall I thought the purpose of this additionalPortlet flag is to add a string that is also added in the windowId of the additional parameters to aid in later parsing.

My thought is that if the switch statement and if statements below in this method do not call url.addParameter() for this portlet, then don't invoke url.addParameter(PARAM_ADDITIONAL_PORTLET, portletWindowIdStr); e.g. add this parameter later if we decide we actually add any parameters for this portlet to the URL.

This would also eliminate passing in the compactParamsForUriFragment property.